libs/transloco/src/lib/tests/scope-resolver.spec.ts

```diff
@@ -73,10 +73,36 @@ describe('ScopeResolver', () => {
         provider: {
           scope: 'nested/scopes/admin-page',
         },
-      })
+      }),
     ).toEqual('nested/scopes/admin-page');
     // one from before
     expect(spy).toHaveBeenCalledTimes(1);
-    expect(spy).toHaveBeenCalledWith('nested/scopes/admin-page', 'nestedScopesAdminPage');
+    expect(spy).toHaveBeenCalledWith(
+      'nested/scopes/admin-page',
+      'nestedScopesAdminPage',
+    );
+  });
+
+  it('should not alter namespace of the scope', () => {
+    resolver = new ScopeResolver({
+      _setScopeAlias: spy,
+      config: {
+        scope: {
+          keepNamespaceCasing: true,
+        },
+      },
+    } as any);
+
+    expect(
+      resolver.resolve({
+        inline: undefined,
+        provider: {
+          scope: 'AdMiN-pAgE',
+        },
+      }),
+    ).toEqual('AdMiN-pAgE');
```

Comment:
WDYT, should we protect from invalid chars?

Reply:
That was the reason why I was using the `toCamelCase` method to handle this.
I do agree `toCamelCase` is not really the place to do this.

But I could create a new method called

```ts
export function removeInvalidChars(str) {
    return str.replace(/\s+|_|-|\//g, '');
}
```

this way we can also alter `toCamelCase` method to this if you like

```ts
export function toCamelCase(str) {
    return removeInvalidChars(str
        .replace(/(?:^\w|[A-Z]|\b\w)/g, (word, index) => index == 0 ? word.toLowerCase() : word.toUpperCase()));
}
```

Reply:
Well, underscores for example are valid, just not part of camel-case. the question is how do you see the usage

Reply:
Hmmm, I think it looked at it the wrong way.

Well as you said in the beginning, it is the responsibility of the consumer to provide a valid `scope` value. 

Of course we could also add a method that would filter out all invalid chars. If you can provide me a list with the allowed `chars` I think I can come up with a regex to only allow these.

Reply:
Let's keep it simple for starters no? it's the consumer's responsibility and if we see that we need to help point to the right direction this kind pf protection can always be added.

Reply:
Yes, indeed.. we can always add extra protection if need be.